### PR TITLE
Healthcare APIs Powershell BYOK Updates

### PR DESCRIPTION
**Pull request by iviark**
## Description

Added support for BYOK for Healthcare Apis

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [X] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/HealthcareApis/HealthcareApis/HealthcareApis.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
   <Import Project="$(MSBuildThisFileDirectory)..\..\Az.props" />
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.HealthcareApis" Version="1.1.0" />
+    <PackageReference Include="Microsoft.Azure.Management.HealthcareApis" Version="3.0.0" />
   </ItemGroup>
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/HealthcareApis.csproj:12]: Just curious if this should be 1.2.0 to match what is in the ChangeLog.md file? Or are they unrelated versions?

File: src/HealthcareApis/HealthcareApis/HealthcareApis/NewAzureRmHealthcareApisService.cs [hunk 2]
```diff
@@ -277,6 +283,16 @@ private Kind ParseKindFromVersion(string fhirVersion)
             return CosmosOfferThroughput;
         }
 
+        private string GetCosmosDBKeyVaultKeyUri()
+        {
+            if (CosmosKeyVaultKeyUri == null)
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/HealthcareApis/NewAzureRmHealthcareApisService.cs:288]: I think the logic in this method works fine, but might be unnecessary, as the default KeyVaultKeyUri value is null.  So, in theory, this method could just do

return CosmosKeyVaultKeyUri;

File: src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md [hunk 3]
```diff
@@ -207,6 +225,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -CosmosKeyVaultKeyUri
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md:228]: Just checking if this should be listed twice? (Once here, once on line 197)

File: src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md [hunk 4]
```diff
@@ -16,23 +16,19 @@ Creates the metadata of a service instance.
...
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
 ## DESCRIPTION
 Creates or updates the metadata of a service instance.
 
 ## EXAMPLES
 
-### Example 1 : Creates a new Azure healthcareapis fhir service named MyService in the resource group MyResourceGroup in a location westus2 with cosmosdb offer throughput = 400
+### Example 1 : Creates a new Azure healthcareapis fhir service named MyService in the resource group MyResourceGroup in a location westus2 with cosmosdb offer throughput = 400 and key vault key uri "https://\<my-keyvault>.vault.azure.net/keys/\<my-key>"
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md:29]: I appreciate you adding an example for this.  As a nit, I'm wondering if it should be in its own example, as it's (currently) not a commonly used feature.  And, its optional and has some challenges for customers to set up the KeyVault correctly.  

I'm wondering if customers, see this in the first example, but don't notice that it's optional, then they will try to use it, which will complicate their setup for a basic Azure API for FHIR service.  So, I'm wondering if we should keep Example 1 simple, add use this CMK feature in a later example.  Debatable.  Your thoughts?

File: src/HealthcareApis/HealthcareApis/help/Set-AzHealthcareApisService.md [hunk 5]
```diff
@@ -46,7 +46,7 @@ Updates an existing healthcareApis fhir service.
 ### Example 1 : Updates the existing healthcareapis service named MyService in the resource group MyResourceGroup  with the cosmosdb OfferThroughput = 500.
 
 ```powershell
-PS C:\> Set-AzHealthcareApisService -Name MyService -ResourceGroupName MyResourceGroup -CosmosOfferThroughput 500
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/help/Set-AzHealthcareApisService.md:49]: Same comment as above.  I wonder if this new parameter should be in a separate, later example.  Thoughts?

File: src/HealthcareApis/HealthcareApis/HealthcareApis/NewAzureRmHealthcareApisService.cs (hunk 2 above)
Comment (iviark) [reply src/HealthcareApis/HealthcareApis/HealthcareApis/NewAzureRmHealthcareApisService.cs:288]: I agree, was just making it consistent with the other code. Will update

Comment (shmartel) [reply src/HealthcareApis/HealthcareApis/HealthcareApis/NewAzureRmHealthcareApisService.cs:288]: Yes, your goal of consistency makes sense.  Arguably, maybe the consistency you gain with this is work the extra code. I can go either way here.  I will leave it to you -- feel free to leave it as is if you prefer it this way.

File: src/HealthcareApis/HealthcareApis/HealthcareApis.csproj (hunk 1 above)
Comment (iviark) [reply src/HealthcareApis/HealthcareApis/HealthcareApis.csproj:12]: Hmm I think making the change log 3.0.0 might make sense, will check to see what Ankita did with PL

Comment (iviark) [reply src/HealthcareApis/HealthcareApis/HealthcareApis.csproj:12]: Updated change log to be equal to this

File: src/HealthcareApis/HealthcareApis/ChangeLog.md [hunk 6]
```diff
@@ -19,6 +19,10 @@
 -->
 ## Upcoming Release
 
+## Version 3.0.0
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/ChangeLog.md:22]: I'm curious about the jump to 3.0.0.
- Changing the first digit indicates a breaking change.  But, I don't think this is a breaking change.
- Changing the second digit indicates a non-breaking addition of functionality (ie, a new feature)

The latter seems most appropriate here.  In that case, 1.2.0 seems like a good version number.  Thoughts here?

File: src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md [hunk 7]
```diff
@@ -16,23 +16,45 @@ Creates the metadata of a service instance.
...
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
 ## DESCRIPTION
 Creates or updates the metadata of a service instance.
 
 ## EXAMPLES
 
-### Example 1 : Creates a new Azure healthcareapis fhir service named MyService in the resource group MyResourceGroup in a location westus2 with cosmosdb offer throughput = 400
+### Example 1 : Creates a new Azure healthcareapis fhir service named MyService in the resource group MyResourceGroup in a location westus2 with cosmosdb offer throughput = 400"
```
Comment (shmartel) [inline src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md:29]: Is this trailing " intentional?

File: src/HealthcareApis/HealthcareApis/ChangeLog.md (hunk 6 above)
Comment (anrudraw) [reply src/HealthcareApis/HealthcareApis/ChangeLog.md:22]: @shmartel 3.0.0 is the nuget package version released for Private Link api version: 2020-03-30. Existing versions in the file seems to be the nuget package versions for the respective api version. I referred below link on before adding this version in the change log. https://www.nuget.org/packages/Microsoft.Azure.Management.HealthcareApis/

File: src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md (hunk 7 above)
Comment (iviark) [reply src/HealthcareApis/HealthcareApis/help/New-AzHealthcareApisService.md]: Accident. fixed it.

File: src/HealthcareApis/HealthcareApis/ChangeLog.md [hunk 8]
```diff
@@ -22,6 +22,7 @@
 ## Version 3.0.0
 * Updated the SDK version to 3.0.0
 * Added support for Private Link
+* Added support for customer managed keys
```
Comment (msJinLei) [inline src/HealthcareApis/HealthcareApis/ChangeLog.md:25]: Please put new item of changelog under upcoming release

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**